x-pack/plugins/code/public/components/main/top_bar.tsx

```diff
@@ -31,20 +33,46 @@ const TopBarContainer = styled.div`
   }
 `;
 
+const SelectContainer = styled(EuiFlexItem)`
+  margin-right: ${theme.euiSizeS};
+`;
+
 interface Props {
   routeParams: MainRouteParams;
   onSearchScopeChanged: (s: SearchScope) => void;
   buttons: React.ReactNode;
+  branches: ReferenceInfo[];
 }
 
-export class TopBar extends React.Component<Props> {
+export class TopBar extends React.Component<Props, { value: string }> {
+  public state = {
+    value: 'master',
```

Comment:
how should we determine what branch it is by the revision?

Reply:
revision is the branch name